execute/exectypes/observation.go

```diff
@@ -18,6 +18,70 @@ type MessageObservations map[cciptypes.ChainSelector]map[cciptypes.SeqNum]ccipty
 // must be encoding according to the destination chain requirements with typeconv.AddressBytesToString.
 type NonceObservations map[cciptypes.ChainSelector]map[string]uint64
 
+// TokenDataObservations contain token data for messages organized by source chain selector and sequence number.
+// There could be multiple tokens per a single message, so MessageTokenData is a slice of TokenData.
+// TokenDataObservations are populated during the Observation phase and depend on previously fetched
+// MessageObservations details and the `tokenDataObservers` configured in the ExecuteOffchainConfig.
+// Content of the MessageTokenData is determined by the tokendata.TokenDataObserver implementations.
+//   - if Message doesn't have any tokens, TokenData slice will be empty.
+//   - if Message has tokens, but these tokens don't require any special treatment,
+//     TokenData slice will contain empty TokenData objects.
+//   - if Message has tokens and these tokens require additional processing defined in ExecuteOffchainConfig,
+//     specific tokendata.TokenDataObserver will be used to populate the TokenData slice.
+type TokenDataObservations map[cciptypes.ChainSelector]map[cciptypes.SeqNum]MessageTokenData
+
+type MessageTokenData struct {
+	TokenData []TokenData
+}
```

Comment:
Why is there a slice of token data per message?

Reply:
Is it a slice of 1 so that you can distinguish between not waiting for token data and waiting for token data?

Reply:
One message can carry multiple tokens that's why we have `[]TokenData`,
```
message seqNr[1234]
     ETH --> [0]TokenData == []bytes{}
     USDC --> [1]TokenData == []bytes{"attestation"}
     GHO --> [2]TokenData == []bytes{}
```

Probably it would be better to rely on the map here, but that will clarify once I start the real implementation. Right now moving it as it is in v1

Reply:
Makes sense! I guess there could be multiple messages for the same token too so a map wouldn't necessarily help much either